pkg/metrics/count_with_timestamp.go

```diff
@@ -0,0 +1,37 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package metrics
+
+// CountWithTimestamp tracks the value of a metric with a given timestamp
+type CountWithTimestamp struct {
+	points  []Point
+	sampled bool
```

Comment:
nit and non-blocking: I'm not sure I understand why we need `sampled` instead of just checking for `points` to be non-empty. Do you know why?